composed-message-processor/composed-message-processor.bal

```diff
@@ -0,0 +1,66 @@
+import ballerina/http;
+
+type InfoRequest record {|
+    string[] states;
+|};
+
+type StateInfo record {|
+    decimal revenue;
+    decimal operatingExpenses;
+    int production;
+    int totalEmployees;
+|};
+
+type AggregratedInfo record {|
+    map<decimal> revenueByState;
+    decimal totalRevenue;
+    decimal maxRevenue;
+    string maxRevenueState;
+    map<decimal> operatingExpensesByState;
+    decimal totalOperatingExpenses;
+    int totalProduction;
+    map<int> productivityByState;
+|};
+
+final map<string> stateRoutes = {
+    Texas: "http://api.texas.office.com.balmock.io",
+    Ohio: "http://api.ohio.office.com.balmock.io",
+    Florida: "http://api.florida.office.com.balmock.io"
+};
+
+AggregratedInfo summary = {
+    revenueByState: {},
+    totalRevenue: 0.0,
+    maxRevenue: 0.0,
+    maxRevenueState: "",
+    operatingExpensesByState: {},
+    totalOperatingExpenses: 0.0,
+    totalProduction: 0,
+    productivityByState: {}
+};
+
+service /api/v1 on new http:Listener(8080) {
+    resource function post state\-info(InfoRequest infoRequest) returns AggregratedInfo|error? {
+        foreach string state in infoRequest.states {
+            string? stateRoute = stateRoutes[state];
+            if stateRoute == () {
+                return error("Invalid state provided");
+            }
+            http:Client stateClient = check new (stateRoute);
```

Comment:
here we have to create stateClients for each request. I suggest lets keep the a set of http:Client values in the map.
```ballerina
final map<http:Client> stateRoutes = {
    Texas: check new ("http://api.texas.office.com.balmock.io"),
    Ohio: check new ("http://api.ohio.office.com.balmock.io"),
    Florida: check new ("http://api.florida.office.com.balmock.io")
};
```

Reply:
Even though there are only three states here, in a practical setting there can be a lot more, then wouldn't it be unfitting to have so many clients created beforehand even if only a few are used? Shouldn't the sample also reflect this?

Reply:
Yes, your point also valid. The frequency of the client usage may differ, but every client will be used. And if the request contain high number of states, this function will spend more time on creating client objects which is not a necessary task of the function ([Single Responsibility Principle](https://stackoverflow.com/questions/77214913/how-to-join-merge-join-two-arrays-in-ballerina/77214914#77214914)).

Reply:
I also prefer to keep the Clients in a map.